### ISSUE TITLE
Better support for rendering CSS and other assets in <head>

### ISSUE THREAD

**Issue by joeldenning** [single-spa/single-spa-layout#77]
Right now, it's a bit complex to render CSS assets for an application. I discussed this with @tsukhu, and we arrived at a proposal for a new way of doing it:

### Part 1 - `<assets>`

Support a new element, `<assets>` that will render the CSS and other assets needed for applications. This would usually be placed in `<head>`.

### Part 2 - renderApplication changes

Support for `renderApplication` to return an object that looks like this:

```js
{
  "assets": `<link rel="stylesheet" href="...">`,
  "content": `<div>...</div>`
}
```

**Comment by joeldenning** [comment]
Here's the file to change: https://github.com/single-spa/single-spa-layout/blob/main/src/server/sendLayoutHTTPResponse.js

We should add a function called `isAssetsNode`, similar to this:

https://github.com/single-spa/single-spa-layout/blob/a440b5f63466ff1dfe4aab7622861b3e70922c99/src/server/sendLayoutHTTPResponse.js#L122

https://github.com/single-spa/single-spa-layout/blob/a440b5f63466ff1dfe4aab7622861b3e70922c99/src/server/sendLayoutHTTPResponse.js#L249-L253

Then we should add `assetsStream` as a new part of the SerializeArgs:

https://github.com/single-spa/single-spa-layout/blob/a440b5f63466ff1dfe4aab7622861b3e70922c99/src/server/sendLayoutHTTPResponse.js#L71

https://github.com/single-spa/single-spa-layout/blob/a440b5f63466ff1dfe4aab7622861b3e70922c99/src/server/sendLayoutHTTPResponse.js#L46-L53

Then inside of `serializeApplication`, we would call `assetsStream.add()` with the assets:

https://github.com/single-spa/single-spa-layout/blob/a440b5f63466ff1dfe4aab7622861b3e70922c99/src/server/sendLayoutHTTPResponse.js#L170

**Comment by joeldenning** [comment]
Resolved in #78
